flake.nix

```diff
@@ -62,159 +62,192 @@
-                '';
-              })
-            ];
-          };
-
-          # F# Environment
-          # `nix develop .#fsharp`
-          fsharp = devenv.lib.mkShell {
-            inherit inputs pkgs;
-            modules = [
-              ({ pkgs, lib, ... }: {
-                packages = with pkgs; [
-                  exercism
-
-                  # .Net
-                  icu
-                  netcoredbg
-                  fsautocomplete
-                  fantomas
-                ];
+            # Erlang shit
+            erlangLatest = pkgs.erlang_26;
+            erlangLibs = getErlangLibs erlangLatest;
 
-                languages.dotnet = {
-                  enable = true;
-                  package = dotnet_8;
-                };
+            # Elixir
+            elixirLibs = getElixirLibs pkgs.elixir-ls;
 
-                enterShell = ''
-                  echo "Starting F# environment..."
-                  exercism version
-                '';
-              })
+            # F#
+            dotnet_8 = with pkgs.dotnetCorePackages; combinePackages [
+              sdk_8_0
             ];
-          };
-
-          # Gleam Environment
-          # `nix develop .#gleam`
-          gleam = devenv.lib.mkShell {
-            inherit inputs pkgs;
-            modules = [
-              ({ pkgs, lib, ... }: {
-                packages = with pkgs; [
-                  exercism
-                ];
 
-                languages.gleam = {
-                  enable = true;
-                };
-
-                enterShell = ''
-                  echo "Starting Gleam environment..."
-                  exercism version
-                '';
-              })
-            ];
-          };
-
-          # Haskell Environment
-          # `nix develop .#haskell`
-          haskell = devenv.lib.mkShell {
-            inherit inputs pkgs;
-            modules = [
-              ({ pkgs, lib, ... }: {
-                packages = with pkgs; [
-                  exercism
+          in
+          {
+            # `nix develop .#ci`
+            # reduce the number of packages to the bare minimum needed for CI
+            ci = pkgs.mkShell {
+              env = mkEnvVars pkgs erlangLatest erlangLibs;
+              buildInputs = with pkgs; [ erlangLatest just rebar3 ];
+            };
+
+            # Rust environemnt support 
+            # nix develop .#rust
+
+            rust = devenv.lib.mkShell
+              {
+                inherit inputs pkgs;
+                modules = [
+                  ({ pkgs, lib, ... }: {
+                    packages = with pkgs; [
+                      rust-analyzer
+                      clippy
+                      rustfmt
+                      cargo-watch
+                      exercism
+                    ];
+
+                    languages.rust = {
+                      enable = true;
+                    };
+
+                    env = mkEnvVars pkgs erlangLatest erlangLibs;
```

Comment:
@AffableMelon You can remove this, I just realized this is some remains from a previous project of mine. I'll remove these weird env settings in a next PR.

Reply:
fixed